src/utils/fhir.js

```diff
@@ -130,11 +137,38 @@ function generateSubscription(code, criteria, url, token) {
  * Get all knowledge artifacts (from servers registered in the
  * db) and save them. Stores all refrenced resources as well.
  */
-function refreshKnowledgeArtifacts() {
+function refreshAllKnowledgeArtifacts() {
+  const servers = db.select('servers', s => s.type === 'KA');
+  servers.forEach(server => refreshKnowledgeArtifact(server));
+}
+
+/**
+ * Get knowledge artifacts from a specific server.
+ *
+ * @param {*} server - the server to refresh artifacts from
+ */
+function refreshKnowledgeArtifact(server) {
+  getResources(server.endpoint, 'PlanDefinition');
+  getResources(server.endpoint, 'Endpoint');
+}
+
+/**
+ * Subscribe to PlanDefinitions on all Knowledge Artifact repos.
+ */
+function subscribeToKnowledgeArtifacts() {
   const servers = db.select('servers', s => s.type === 'KA');
-  servers.forEach(server => {
-    getResources(server.endpoint, 'PlanDefinition');
-    getResources(server.endpoint, 'Endpoint');
+  servers.forEach(async server => {
+    // TODO: generate access token for subscription
+    const subscription = generateSubscription(
+      'PlanDefinition?_lastUpdated=gt2021-01-01',
+      `${process.env.BASE_URL}/notif/ka/${server.id}`,
+      'admin'
+    );
+    const token = await getAccessToken(server.endpoint);
+    const headers = { Authorization: `Bearer ${token}` };
+    axios
+      .post(`${server.endpoint}/Subscription`, subscription, { headers: headers })
```

Comment:
Did we want to create some kind of unique id and do a `axios.put` for these subscriptions also so that subscriptions don't keep getting created on the KA server?

Reply:
Updated to be a PUT and created an id in the form `sub{KA Server ID}`. Took some of your id changes and copied them in here too.